File: Easy/283. Move Zeroes/main.cpp
#include <iostream>
#include <string>
#include <vector>
#include <unordered_map>


using namespace std;

class Solution {
public:
    void moveZeroes(vector<int>& nums) {

        int left = 0;
        for (int right = 0; right < nums.size(); right++)
        {
            if (nums[right] != 0)
            {
                swap(nums[right], nums[left]);
                left++;
            }
        }

    }

};



int main() {


    Solution s = Solution();

    vector<int> t1 = { 0,1,0,3,12 };
    vector<int> t2 = { 1,2 };

    s.moveZeroes(t1);

    for (auto&& i : t1)
    {
        cout << i << endl;
    }



    return 0;
}

File: Easy/283. Move Zeroes/main.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        
        idx_left=0
        for right_idx in range(len(nums)):
            if nums[right_idx]!=0:
                temp= nums[right_idx]
                nums[right_idx]=nums[idx_left]  
                nums[idx_left]=temp
                idx_left+=1       
            


        
nums1 = [0,1,0,3,12]
s=Solution()
s.moveZeroes(nums1)
print(nums1)
